File: src/test/java/com/ddd/trainings/idea/domain/TrainingIdeaBuilder.java
package com.ddd.trainings.idea.domain;

import com.ddd.trainings.idea.infra.TrainingIdeaRepo;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class TrainingIdeaBuilder {
    private IdeaId ideaId = new IdeaId("test-id");
    private TrainerId trainerId = TrainerId.from("trainer-id");
    private IdeaNameVO name = IdeaNameVO.from("Test Training");
    private TrainingDurationVO duration = TrainingDurationVO.from(2);

    private final TrainingIdeaRepo ideaRepo;

    public TrainingIdeaBuilder withIdeaId(IdeaId ideaId) {
        this.ideaId = ideaId;
        return this;
    }

    public TrainingIdeaBuilder withTrainerId(TrainerId trainerId) {
        this.trainerId = trainerId;
        return this;
    }

    public TrainingIdeaBuilder withName(IdeaNameVO name) {
        this.name = name;
        return this;
    }

    public TrainingIdeaBuilder withDuration(TrainingDurationVO duration) {
        this.duration = duration;
        return this;
    }

    public TrainingIdea build() {
        return new TrainingIdea(ideaId, trainerId, name, duration);
    }

    public TrainingIdea inDb() {
        return ideaRepo.save(build());
    }
} 

File: src/Ddd.Tests/Trainings/Idea/Domain/TrainingIdeaBuilder.cs
using Ddd.Trainings.Idea.Domain.Ports;
using Ddd.Trainings.Proposal.Domain;

namespace Ddd.Trainings.Idea.Domain.Tests;

public class TrainingIdeaBuilder
{
    private IdeaId _ideaId = new IdeaId("test-id");
    private TrainerId _trainerId = TrainerId.From("trainer-id");
    private IdeaNameVO _name = IdeaNameVO.From("Test Training");
    private TrainingDurationVO _duration = TrainingDurationVO.From(2);
    private bool _isProposed;

    private readonly ITrainingIdeaRepo _ideaRepo;

    public TrainingIdeaBuilder(ITrainingIdeaRepo ideaRepo)
    {
        _ideaRepo = ideaRepo;
    }

    public TrainingIdeaBuilder WithIdeaId(IdeaId ideaId)
    {
        _ideaId = ideaId;
        return this;
    }

    public TrainingIdeaBuilder WithTrainerId(TrainerId trainerId)
    {
        _trainerId = trainerId;
        return this;
    }

    public TrainingIdeaBuilder WithName(string name)
    {
        _name = IdeaNameVO.From(name);
        return this;
    }

    public TrainingIdeaBuilder WithDuration(TrainingDurationVO duration)
    {
        _duration = duration;
        return this;
    }

    public TrainingIdeaBuilder Proposed(bool isProposed)
    {
        _isProposed = isProposed;
        return this;
    }

    public TrainingIdea Build()
    {
        var idea = new TrainingIdea(_ideaId, _trainerId, _name, _duration);
        if (_isProposed)
        {
            idea.Propose(new FakeReviewerPolicy());
        }
        return idea;
    }

    public TrainingIdea InDb()
    {
        return _ideaRepo.Save(Build());
    }

    private class FakeReviewerPolicy : IReviewerPolicy
    {
        public ReviewerId SelectReviewer()
        {
            return new ReviewerId("fake-reviewer-id");
        }
    }
} 